client/src/schema/repository.ts

```diff
@@ -132,7 +133,23 @@ export class RestRepository<Data extends BaseData, Record extends XataRecord = D
     };
   }
 
-  async create(object: Data): Promise<Record> {
+  async create(object: Data): Promise<Record>;
+  async create(recordId: string, object: Data): Promise<Record>;
+  async create(a: string | Data, b?: Data): Promise<Record> {
+    if (isString(a) && isObject(b)) {
+      if (a === '') throw new Error("The id can't be empty");
+      return this.#insertRecordWithId(a, b);
+    } else if (isObject(a) && isString(a.id)) {
+      if (a.id === '') throw new Error("The id can't be empty");
```

Comment:
```suggestion
      if (!a.id) throw new Error("The id can't be empty, null or undefined");
```

The test will need to be updated.

Reply:
`isString(a.id)` checks for null and undefined already

Reply:
Cool